distributed/tests/test_client.py

```diff
@@ -3240,6 +3240,63 @@ def test_default_get(loop_in_thread):
         assert dask.base.get_scheduler() == pre_get
 
 
+@gen_cluster(config={"scheduler": "sync"}, nthreads=[])
+async def test_get_scheduler_default_client_config_interleaving(s):
```

Comment:
This test is copied from dask/dask but since this functionality is so deeply entangled I think this is worth it